File: src/문자열/BOJ_2857_FBI.js
const fs = require("fs");
const stdin = (process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString()
    : `47-FBI
BOND-007
RF-FBI18
MARICA-13
13A-FBILL`
).split("\n");

const input = (() => {
    let line = 0;
    return () => stdin[line++];
})();

let idx = 0;
let answer = [];
let regex = /FBI/g;

while (idx++ < 5) {
    let str = input();
    if (str.search(regex) != -1) answer.push(idx);
}

console.log(answer.length == 0 ? "HE GOT AWAY!" : answer.join(" "));

File: src/문자열/BOJ_2857_FBI.java
package 문자열;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class BOJ_2857_FBI {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder answer = new StringBuilder();
        Pattern pattern = Pattern.compile("FBI");
        int idx = 0;

        while (idx++ < 5) {
            Matcher matcher = pattern.matcher(br.readLine());
            if (matcher.find()) answer.append(idx + " ");
        }

        System.out.println(answer.length() == 0 ? "HE GOT AWAY!" : answer);
    }
}
